patches/api/0371-Custom-Potion-Mixes.patch

```diff
@@ -0,0 +1,157 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 7 Oct 2021 14:34:59 -0700
+Subject: [PATCH] Custom Potion Mixes
+
+
+diff --git a/src/main/java/io/papermc/paper/potion/PotionMix.java b/src/main/java/io/papermc/paper/potion/PotionMix.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..9a9c28254bd1c0ec540a38635a5d8e630d77eaa7
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/potion/PotionMix.java
+@@ -0,0 +1,78 @@
++package io.papermc.paper.potion;
++
++import org.bukkit.Keyed;
++import org.bukkit.NamespacedKey;
++import org.bukkit.inventory.ItemStack;
++import org.bukkit.inventory.RecipeChoice;
++import org.jetbrains.annotations.NotNull;
++
++import java.util.Objects;
++
++public class PotionMix implements Keyed {
```

Comment:
final?

Reply:
I guess it can be final now, but it probably wont be final later on when support for the vanilla mixes is added. itll be like the ShapelessRecipe and CraftShapelessRecipe

Reply:
I don't see why a vanilla recipe couldn't be represented by PotionMix as is

Reply:
I think just to prevent the conversion back and forth between internal and api types